test/unit/lib/runner/writers/action-json-writer-tests.js

```diff
@@ -0,0 +1,170 @@
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'modal',
+                      action: 'click',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+          {
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'modal',
+                      action: 'click',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        ],
+      };
+
+      fs = {
+        writeFileSync: sinon.stub(),
+        readFileSync: sinon.stub(),
+      };
+
+      configHandler = {
+        get: sinon.stub(),
+      };
+
+      obj = {};
+
+      mockery.registerMock('fs', fs);
+      mockery.registerMock('path', path);
+      mockery.registerMock('../../util/config-handler.js', configHandler);
+
+      writeActionReportToDisk = require('../../../../../lib/runner/writers/action-json-writer.js');
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    it('should call path.resolve with the configs testPath', function() {
+      let report = {
+        testReports: [],
+      };
+      configHandler.get.returns('./testPath');
+
+      writeActionReportToDisk(report);
+
+      expect(path.resolve.args).to.deep.equal([
+        [
+          './testPath',
+        ],
+      ]);
+    });
+
+    describe('for each test report of the passed in report', function() {
+      it('should call fs.readFileSync with the path to the test and \'utf8\'', function() {
+        path.resolve.returns('./testpath');
+        fs.readFileSync.returns('{}');
+
+        writeActionReportToDisk(report);
+
+        expect(fs.readFileSync.args).to.deep.equal([
+          [
+            './testpath/testOne',
+            'utf8',
+          ],
+          [
+            './testpath/testOne',
+            'utf8',
+          ],
+        ]);
+      });
+
+      it('should assign customReport', function() {
+        let customReport = {
+          'name': `test`,
+      };
+        path.resolve.returns('./testpath');
+        fs.readFileSync.returns('{}');
+
+        writeActionReportToDisk(report);
+
+        expect(customReport.name).to.equal('test');
+      });
+    });
+
```

Comment:
You should have a section of tests regarding the actions, and that mutiple custom reports are created and then the only way to check that is when the report is used, again with fs.writeFileSync at the end of the function